File: top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
      Map<Integer, Integer> map = new HashMap<>();
      ArrayList<Integer> list = new ArrayList<Integer>();      
        int i = 0;
        while(i< nums.length){
            map.put(nums[i], map.getOrDefault(nums[i], 0) + 1);
            i++;
        }
        for ( int key : map.keySet() ) {           
                list.add(map.get(key));
            }
       Collections.sort(list,  Collections.reverseOrder());
        int[] arr = new int[k];
        int add = 0;
      while(add < k){
        for(Map.Entry<Integer, Integer> entry: map.entrySet()) {
                if(entry.getValue() ==list.get(add)) {
                    arr[add] = entry.getKey();
                    }
                  }
               map.remove(arr[add]);
               add++; 
             }
     return arr;
    }
}

File: top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = Counter(nums)
        unique = list(freq.keys())    
        n = len(unique)
        k = n-k 
        def quickSelect(l,r): 
            pivot = unique[r] 
            i = l 
            for j in range(l,r): 
                if freq[unique[j]]<=freq[pivot]: 
                    unique[i],unique[j]=unique[j],unique[i]
                    i+=1 
            unique[i],unique[r]=unique[r],unique[i] 
            if i>k: 
                return quickSelect(l,i-1)
            elif i<k: 
                return quickSelect(i+1,r)
            else:
                return unique[i]
        quickSelect(0,n-1)
        
        return unique[k:]